pulpcore/pulpcore/app/models/repository.py

```diff
@@ -373,12 +382,15 @@ def add_content(self, content):
         """
         if self.complete:
             raise ResourceImmutableError(self)
-        with suppress(IntegrityError):
-            association = RepositoryContent(
-                repository=self.repository,
-                content=content,
-                version_added=self)
-            association.save()
+
+        if self.content_exists(content):
+            raise DuplicateContentException(content, self.repository)
```

Comment:
Do we really want to raise an exception here?  Perhaps add_content() should be idempotent.